chainerio/filesystems/hdfs.py

```diff
@@ -84,14 +84,27 @@ def _run_klist(use_keytab=False):
 
 class HdfsFileSystem(FileSystem):
     def __init__(self, io_profiler=None, root=""):
+        self.username = self._get_principal_name()
```

Comment:
Why is this moved to before superclass constructor? How is this related to default handlers?

Reply:
The `self.username` is used in the setter of `root` at https://github.com/chainer/chainerio/blob/dfeaa0161c4840c68a713fb4b0f928c132e146c3/chainerio/filesystems/hdfs.py#L106, which is overridden at https://github.com/chainer/chainerio/blob/dfeaa0161c4840c68a713fb4b0f928c132e146c3/chainerio/filesystems/hdfs.py#L102 and called in the superclass constructor
https://github.com/chainer/chainerio/blob/dfeaa0161c4840c68a713fb4b0f928c132e146c3/chainerio/io.py#L33